x/uibc/uics20/memo_handler.go

```diff
@@ -58,13 +58,20 @@ func (mh *MemoHandler) onRecvPacketPrepare(
 		return nil, sdkerrors.Wrap(err, "can't parse ftData.Receiver bech32 address")
 	}
 
+	var memoPayload []byte
 	if strings.EqualFold(ftData.Sender, gmp.DefaultGMPAddress) {
 		events = append(events, "axelar GMP transaction")
 		mh.isGMP = true
-		return events, nil
+		gh := gmp.NewHandler()
+		gmpMessage, err := gh.OnRecvPacket(*ctx, mh.received, mh.memo, mh.receiver)
+		if err != nil {
+			return events, err
```

Comment:
1. let's log an error here
2. unmarshalling error nor validation error should not abort the transfer. Please update a code and add a test for that.